app/models/miq_expression.rb

```diff
@@ -1775,15 +1775,9 @@ def self.atom_error(field, operator, value)
                                                                                     :error_message => err.message}
       end
       return false
-    when :ruby
-      return _("Ruby Script must not be blank") if value.blank?
-      return false
```

Comment:
according to @gtanzillo - yes. Reason behind it - there could be datatypes which are not analyzed.   I think if datatype is not in our list of 'acceptable' datatypes than we should treat it as :string, but may be it should be for another PR